### ISSUE TITLE
Spike: Do weverify the image digest and does the reference actually exists

### ISSUE THREAD

**Issue by LappleApple** [kubernetes-sigs/promo-tools#1131]
## Objective

- [ ] Support the goal of breaking up the image promoter monolith by investigating the questions raised below

## Questions to answer

- [ ] Check if we verify the image digest and the reference actually exists, to evaluate:
      - Did we create an image using appropriate tools and process?
      - Would scanning for CVEs be a blocker? 
      - If yes, could we require provenance?
      - Would infra in staging help make it faster than a complete image push/pull? Not in place today.
      - How could we ensure that the SHA is valid?
      - How do we scan the image for any high or critical CVE?

**Note:** Be sure to actively seek other questions from members of SIG Release and other relevant SIGs as part of your research.

## Steps
- [ ] Present a 1-2 page proposal describing the necessary implementation steps and listing pros/cons/tradeoffs
- [ ] Seek input from SIG members and achieve buy-in so the group can reach consensus and move forward

## Context and things to think about while working on this task
- Relates to the "Validating signatures from staging in parallel" workflow step
- We currently have no way to verify the SHA of an image. 
- We delegate responsibility for establishing which image to put in prod but we don't verify the image provenance--coming from stg container registry. 
- We know the source of an image, but we don't verify what is published. 
- Any maintainer could do the promotion. Nothing prevents maintainer w/access to staging repo from pushing something that could impact the registry.

**Comment by saschagrunert** [comment]
Closing in favor of https://github.com/kubernetes-sigs/promo-tools/issues/1701, thank you for all the work which went into this. It's consolidated into the single issue.
